README.md

```diff
@@ -1,3 +1,191 @@
-# strimzi-kafka-oauth
+Strimzi Kafka OAuth Guide
+=========================
+
+
+Strimzi Kafka OAuth modules provide support for using OAuth2 as authentication mechanism when establishing a session with Kafka Broker.
+
+
+OAuth2 for Authentication
+-------------------------
+
+One of the advantages of OAuth2 compared to direct client-server authentication is that client credentials are never shared with the server. Rather, 
+there exists a separate OAuth2 authorization server that application clients and application servers communicate with. This way user management, 
+authentication, and authorization is 'outsourced' to authorization server.  
+
+User authentication is separated into an outside step which user manually performs to obtain an access token or a refresh token, which grants a 
+limited set of permissions to a specific client app. In the simplest case, the client application can authenticate in its own name using client 
+credentials. While the client secret is in this case packaged with application client, the benefit is still that it is not shared with application 
+server (Kafka Broker in our case) - the client first performs authentication against OAuth2 authorization server in exchange for an access token, 
+which it then sends to authorization server instead of its secrets. Access tokens can be independently tracked and revoked at will, and represent a 
+limited access to resources on application server.
+
+When user authenticates and authorizes application client in exchange for a token, the client application can be packaged with only the access token
+or the refresh token. User's username and password are never packaged with application client. The access token is sent directly to Kafka Broker 
+during session initialisation, while a refresh token is first used to ask authorisation server for a new access token, before sending it to Kafka 
+Broker to start a new authenticated session.
```

Comment:
I think a diagram showing this interaction would really help people understand. I'm thinking of a directed graph with the server, client and authorization server as nodes and with edges indicating a message. Edges annotated with a number, which corresponds to a description in the text.

Reply:
This is a very general, well specified behaviour covered by OAuth2 spec and widely covered on the Web. Such a diagram should be part of Keycloak project docs or we find one appropriate somewhere else, and link to it?

Reply:
I'm open to whatever is easiest (so long as it's compatible with the license of course).